components/director/internal/domain/formationconstraint/operators/redirect_notification_operator.go

```diff
@@ -0,0 +1,63 @@
+package operators
+
+import (
+	"context"
+	"runtime/debug"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/formationconstraint"
+	"github.com/kyma-incubator/compass/components/director/pkg/log"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// RedirectNotificationOperator represents the redirect notification operator
+	RedirectNotificationOperator = "RedirectNotification"
+)
+
+// NewRedirectNotificationInput is input constructor for DestinationCreatorOperator. It returns empty OperatorInput
+func NewRedirectNotificationInput() OperatorInput {
+	return &formationconstraint.RedirectNotificationInput{}
+}
+
+// RedirectNotification is an operator that based on different condition could redirect the formation assignment notification
+func (e *ConstraintEngine) RedirectNotification(ctx context.Context, input OperatorInput) (bool, error) {
+	log.C(ctx).Infof("Starting executing operator: %s", RedirectNotificationOperator)
+
+	defer func() {
+		if err := recover(); err != nil {
```

Comment:
This defer(+ the same defer in the DC operator) are just for safety because if there is something wrong in the "retrieveFromMemoryAddress" we could get an object that from the type system PoV is graphql.Webhook but under the hood to hold properties/fields from another structure, e.g. model.Webhook. And accessing some of the fields could result in an error. In summary, we have it to prevent unexpected errors while working with memory addresses.